tests/Unit/Core/Form/IdentifiableObject/CommandBuilder/Product/ShippingCommandBuilderTest.php

```diff
@@ -31,27 +31,92 @@
 use Generator;
 use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductShippingCommand;
 use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\DeliveryTimeNoteType;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product\ShippingCommandsBuilder;
 
 class ShippingCommandBuilderTest extends AbstractProductCommandBuilderTest
 {
+    private const MODIFY_ALL_SHOPS_PREFIX = 'modify_all_shops_';
+
     /**
-     * @dataProvider getExpectedCommands
+     * @dataProvider getExpectedSingleShopCommands
      *
      * @param array $formData
      * @param array $expectedCommands
      */
-    public function testBuildCommand(array $formData, array $expectedCommands): void
+    public function testBuildSingleShopCommands(array $formData, array $expectedCommands): void
     {
-        $builder = new ShippingCommandsBuilder();
-        $builtCommands = $builder->buildCommands($this->getProductId(), $formData);
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
         $this->assertEquals($expectedCommands, $builtCommands);
     }
 
+    /**
+     * @dataProvider getExpectedAllShopCommands
+     *
+     * @param array $formData
+     * @param array $expectedCommands
+     */
+    public function testBuildAllShopCommands(array $formData, array $expectedCommands): void
+    {
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
+        $this->assertEquals($expectedCommands, $builtCommands);
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getExpectedAllShopCommands(): Generator
+    {
+        $command = $this->getAllShopsCommand();
+        /* @phpstan-ignore-next-line */
+        $command->setCarrierReferences(['1', '2', '3']);
```

Comment:
Why not use integers here? It makes sense to provide string in the form data but we should ensure that the builder correctly cast them no?
That said the builder component doesn't handle casting in arrays 🤔 Then my suggestion to use `CarrierId` actually makes more sense with this in mind

Reply:
yep, my bad. My intention indeed was only to provide strings as the form data. Ill leave them as integers now, but it probably would be nice having Id classes to be safe.